packages/dm-core-plugins/src/job/JobPlugin.tsx

```diff
@@ -44,43 +45,39 @@ export const JobPlugin = (props: IUIPlugin & { config: JobPluginConfig }) => {
   }: { config: JobPluginConfig; idReference: string } = props
   const DmssApi = useDMSS()
 
-  const jobTargetAddress = (): string => {
-    if (config.jobTargetAddress.addressScope !== 'local') {
+  const jobTargetAddress = useMemo((): string => {
+    if ((config.jobTargetAddress.addressScope ?? 'local') !== 'local') {
       return config.jobTargetAddress.targetAddress
     }
     if (['self', '.'].includes(config?.jobTargetAddress.targetAddress)) {
       return idReference
     }
     return idReference + config.jobTargetAddress.targetAddress
-  }
+  }, [config])
 
-  const jobInputAddress = (): string => {
-    if (config.jobInput.addressScope !== 'local') {
+  const jobInputAddress = useMemo((): string => {
+    if ((config.jobInput.addressScope ?? 'local') !== 'local') {
       return config.jobInput.targetAddress
     }
     if (['self', '.'].includes(config.jobInput.targetAddress)) {
       return idReference
     }
     return idReference + config.jobInput.targetAddress
-  }
+  }, [config])
   const { tokenData } = useContext(AuthContext)
   const username = tokenData?.preferred_username
 
-  const [jobEntityId, setJobEntityId] = useState<string>('')
-  const [jobId, setJobId] = useState<string | undefined>(undefined)
   const [jobExists, setJobExists] = useState(false)
   const [result, setResult] = useState<GetJobResultResponse | null>(null)
-  const [allowStartJob, setAllowJobStart] = useState(false)
-
   const {
-    start,
-    error,
-    fetchResult,
-    fetchStatusAndLogs,
-    logs,
-    status,
-    remove,
-  } = useJob(jobEntityId, jobId)
+    document: jobDocument,
+    isLoading,
```

Comment:
`isLoading: bool`

Reply:
not really needed? and the type would be `Boolean` ;)